blueflood-core/src/main/java/com/rackspacecloud/blueflood/outputs/handlers/RollupHandler.java

```diff
@@ -53,20 +60,51 @@
     protected final Histogram numHistogramPointsReturned = Metrics.histogram(RollupHandler.class, "Histogram points returned");
 
     private static final boolean ROLLUP_REPAIR = Configuration.getInstance().getBooleanProperty(CoreConfig.REPAIR_ROLLUPS_ON_READ);
+    protected final DiscoveryIO discoveryHandler = loadDiscoveryModule();
```

Comment:
We have this cruft code in a lot of places, which loads the discovery modules. May be we should have a separate class which does this for us or something.